File: frontend_ghotel/src/app/components/models/reservation.ts
import { Client } from '../models/client';
import { LigneReservation } from '../models/ligne-reservation';
import { Facture } from './facture';

export class Reservation {
  id?: number;
  code!: string;
  dateReservation!: Date;
  dateDebut!: Date;
  dateFin!: Date;
  montantTotal!: number;
  montantRemise!: number;
  statut!: 'EN_COURS' | 'CONFIRMEE' | 'ANNULEE'; // ou string si libre
  client_id!: number;
  client!: Client;
  lignes!: LigneReservation[];
  showDetails?: boolean;
  facture!: Facture | null;
}

File: backend_ghotel/src/main/java/com/ikadjate/backend/model/Reservation.java
package com.ikadjate.backend.model;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.ikadjate.backend.dto.LigneReservationDto;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Reservation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(nullable = false)
    private String code;
    @Column(nullable = false)
    private LocalDate dateReservation;
    @Column(nullable = false)
    private LocalDate dateDebut;
    @Column(nullable = false)
    private LocalDate dateFin;
    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private StatutReservation statut;
    @Column(nullable = false)
    private BigDecimal montantTotal;

    @ManyToOne(optional = false)
    @JoinColumn(name = "client_id", nullable = false)
    private Client client;
    
    @OneToMany(mappedBy = "reservation", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    private List<LigneReservation> lignes;

    @OneToOne(mappedBy = "reservation", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonBackReference // ← Pour éviter la récursivité infinie
    private Facture facture;
    
}
